### PR TITLE
Meteogram component refactor

### PR DESCRIPTION
**Pull request by ruchimotwaniBC**

### CODE REVIEW

File: doors_dashboards/components/meteogram.py [hunk 1]
```diff
@@ -208,18 +229,94 @@ def register_callbacks(self, app: dash.Dash, component_ids: List[str]):
...
+            State(COMPONENT_STORE_ID, 'data'),
+            prevent_initial_call=True
+        )
+        def datepicker_to_component_store(temp_data, component_data):
+            if temp_data is None:
+                return  no_update
+            if component_data is not None:
+                component_data = component_data or {}
+            else:
+                component_data = {}
```
Comment (TonioF) [inline doors_dashboards/components/meteogram.py:320]:
```suggestion
            component_data = component_data or {}
```

This is doing the same thing in just one line.

File: doors_dashboards/components/scattermap.py [hunk 2]
```diff
@@ -166,19 +168,29 @@ def update_general_store_after_station_selection(
...
+                        group = gdf.iloc[0][group_name]
+                        general_data["groups"] = {}
+                        general_data["groups"][collection_name] = (
+                            group)
             if "variable" not in general_data:
                 general_data["variable"] = {}
             general_data["variable"][collection_name] = (
                 self.feature_handler.get_default_variable(collection_name))
             return general_data
+
```
Comment (TonioF) [inline doors_dashboards/components/scattermap.py:196]:
```suggestion
```
Only one empty line at the end, please.

File: doors_dashboards/components/scattermap.py [hunk 3]
```diff
@@ -166,19 +168,29 @@ def update_general_store_after_station_selection(
...
+                        general_data["meteogram_data"] = {
+                            'lon': lon,
+                            'lat': lat,
+                            'label': text
+                        }
+                    else:
+                        group = gdf.iloc[0][group_name]
+                        general_data["groups"] = {}
+                        general_data["groups"][collection_name] = (
+                            group)
```
Comment (TonioF) [inline doors_dashboards/components/scattermap.py:190]:
```suggestion
                    text = click_data['points'][0]['text']
                    general_data["selected_point"] = {
                        'lon': lon,
                        'lat': lat,
                        'label': text
                    }
                    if group_name:
                        group = gdf.iloc[0][group_name]
                        if "groups" not in general_data:
                            general_data["groups"] = {}
                        general_data["groups"][collection_name] = (group)
```
I do not understand why you update this differently whether groups are provided or not. Why not do both always?
Please note I changed meteogram_data to selected_point, as this way it is not as commited to another component.

File: doors_dashboards/components/meteogram.py [hunk 4]
```diff
@@ -208,18 +227,79 @@ def register_callbacks(self, app: dash.Dash, component_ids: List[str]):
 
         @app.callback(
             Output(METEOGRAM_IMAGE_ID, 'children'),
-            [Input("scattermap", 'clickData'),
-             Input(METEOGRAM_DATE_PICKER_ID, 'date'),
-             Input(METEOGRAM_CHOOSER_ID, 'value')]
+            Input("general", "data"),
+            Input(COMPONENT_STORE_ID, 'data'),
+            State(COMPONENT_STORE_ID, 'data')
```
[the remark this answers is not in the document]
Comment (ruchimotwaniBC) [reply doors_dashboards/components/meteogram.py:237]: I had to pass Input because it was not working with only state.

**Pull request merged**